cmd/fdo_client/deviceinfo_linux.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"log/slog"
+	"os"
+	"strings"
+)
+
+func getDeviceInfo() (string, error) {
+	deviceInfo, err := getDeviceInfoFromEnv()
+	if err == nil {
+		return deviceInfo, nil
+	}
+	if diDeviceInfo == deviceInfoSerialNumber {
```

Comment:
The only use for the `-di-device-info` flag value is to check it against the sentinel value `deviceInfoSerialNumber` or use it as the network interface name to then get the MAC address for the actual DI device info.

1. The sentinel value of `"serial-number"` is not any better than `""` and can thus be removed.
2. The description of the flag is `"either serial-number or mac-address's iface e.g. eth0"` which should be clarified to mean that you can't actually specify a serial number via flag (only env var).
3. Should you only be able to specify the serial number by env var? Seems like setting the flag should be an option.
4. The interface name logic seems too conflated. I think that should move to a separate flag, like `-di-device-info-mac` or something.

Reply:
so we could have 

`-di-device-info`, defaults to an empty string, meaning it'll go and check, on linux, on sysfs to try and gather the serial number. If not empty, it'll be used straight away in the device credentials

`-di-device-info-mac=iface`, defaults to empty and it's not used. If non empty, it'll be used instead of the above (this is what I don't like)

what do you think

Reply:
Yeah, that's basically what I was thinking. Yes, the problem is "what if both flags are set?" That's kind of an invalid state. Either we can error, saying that the flags are mutually exclusive, or give one priority and log a warning. If we give one priority, I'd actually give it to the `di-device-info` flag, since that one is more explicit: "this is the device info to use."

Reply:
I’d go with erroring out if both are set - that way intentions are clearer and it’ll be immediately visible to the user (instead of being an hidden decision from us to give one priority over the other)

Reply:
comment addressed as per this conversation